src/foam/dao/FileJournal.js

```diff
@@ -298,21 +303,18 @@ foam.CLASS({
       javaCode: `
         try {
           String line = reader.readLine();
-          if ( line == null )
-            return null;
+          if ( line == null ) return null;
+          StringBuilder sb = new StringBuilder();
+          sb.append(line);
           if ( getMultiLine() ) {
-            StringBuilder sb = new StringBuilder();
-            while ( ! line.equals("})") ) {
+            while( ! line.trim().endsWith("})") && ! line.trim().startsWith("//") ) {
```

Comment:
I don't agree with doing .trim().

This means that any comments inside the script will be lost.  Also any instance of }) will end the script early.

We also shouldn't build the StringBuilder before checking getMultiline() otherwise we'll be building a new Stringbuilder for every line in single-lined journals, that's wasteful.

I think what you want for this method is:

```
String line = reader.readLine();
if ( ! getMultiline() ) return line;
if ( line == null ) return null;
if ( line.endsWith("})") return line;

StringBuilder sb = new StringBuilder();
sb.append(line);

do {
  line = reader.readLine();
  if ( line == null ) return null;
  sb.append(line);
  sb.append("\n");
} while ( ! line.equals("})") );

return sb.toString();
```

In the non-multiline case this returns quickly with no extra work than before.

If we're in a multiline journal but the entry appears to be a single line entry `endsWith("})")` then we just return the line as is.

If we're in a multiline journal and the entry appears to be a multi-line entry (does not end with "})") then we buffer lines until we see a line that only consists of "})".

I don't think you need to escape newlines in the buffer, just insert real new-lines, the StringParser should handle them just fine.